src/providers/TxHistoryProvider.tsx

```diff
@@ -0,0 +1,134 @@
+import { TransactionResponse } from '@ethersproject/providers'
```

Comment:
TxHistoryProvider is the crux of this PR

- maintains state of pending/historical transactions. provides `transactions` via context
- uses a poller to periodically update pending TXs
- provides addTransaction/removeTransaction functions via context for manipulating state